### DISCUSSION TITLE
start a new mindmap based on a node with links from one to the other

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#662] [Ideas]
I would like to be able to right+click a node, and among the choices be one to "start a new mindmap" that is based on the selected node.  The new mindmap would have a suggested title of its starting node. And there would be links going between the starter node in the first mindmap and the root node in the new mindmap.

I think this would be very helpful for when I am creating and managing projects. The projects would be listed in the first mindmap. The projects would be contained in their own mindmaps.

This is sort of like how wiki's work, where you create a node by calling it by name in an existing node.

I wonder if others would find it useful.

~ Bal

**Comment by euu2021** [comment]
Please, try `File->Move branch to new map…`. Seems to be exactly what you want.

**Reply by wordmuse** [reply]
Interesting and close - but not quite.  I really just want the one node copied - not necessarily its children.  In practical terms, a child on the original map might just be a summary explanation of what the destination map contains. Or it could contain a skeleton of what I might want on the destination map. The point is that moving the branch defeats this function.

Also - the link from the root of the new map back to the original map is not node to node. In other words, it's not a direct link to the originating node. It is just a link back to the original map. For this to be useful to me, I need the new root to link directly to the originating root.

I didn't know about this "move branch to new map" function. I can see other uses for that. So thank you!

~ Bal

**Reply by euu2021** [reply]
> I really just want the one node copied - not necessarily its children.

You can achieve that with those simple steps:

1. select the node of interest (the node that you want to be the root of the new map)
2. `Navigate->Select visible branch (Ctrl+A)`
3. unselect only the node of interest (Ctrl + click on it)
4. `Edit->Cut (Ctrl+X)`
5. `File->Move branch to new map…`
6. `Edit->Paste`

One advantage of this is that you can, on step 3, also unselect other nodes that you to be moved along to the new map.

EDIT: just to be sure that you understand, I will add that, before step 1, you need to unfold, at least, one level of the node of interest.

**Reply by wordmuse** [reply]
OK - that gets me closer. So thanks. But the link back to the original node is missing.  There is a workaround for that, too.  

I'm asking for this without the workarounds. But given that there are simple workarounds, I can see the priority for doing this being low. I am glad to know that the workarounds exist! Thanks!

~ Bal

**Reply by macmarrum** [reply]
Hi Bal

I also need this functionality from time to time and I had intended to create a script but never got round to it. You've reminded me of this :+1:

Here's the script: https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/startNewMindmap.groovy
As usual, just save it to `<user-dir>/scripts`, restart Freeplane and you'll find it in `right-click (on a node)->Scripts->Start New Mindmap`

Let me know how it works for you.

Mac

**Reply by wordmuse** [reply]
Just saw this now. Testing it - it works perfectly! Thank you so much, Mac!

Best Regards,
~ Bal
